File: src/test/scala/uk/co/halfninja/randomnames/EnglishlikeNameGeneratorTest.scala
package uk.co.halfninja.randomnames

import org.scalatest._
import org.scalatest.matchers.ShouldMatchers._

class EnglishlikeNameGeneratorTest extends FunSuite {

	/**
	 * Not actually testing anything here.. just printing things out.
	 * Fun!
	 */
	test("Generation of names works") {
		val generator = NameGenerators.standardGenerator()
		(1 to 10).foreach { i =>
			println(generator.generate(Gender.male))
			println(generator.generate(Gender.female))
		}
	}

	test("Seeded name generation returns expected names") {
		val generator = NameGenerators.standardGenerator()
		(1 to 10).foreach { i =>
			generator.generate(Gender.male, 1234567).toString() should be ("Matthew Baker")
			generator.generate(Gender.male, 1234568).toString() should be ("Sam Greenton")
			generator.generate(Gender.male, 1234569).toString() should be ("Richard Robertsford")
			generator.generate(Gender.male, 1234570).toString() should be ("Jack Patel")
			generator.generate(Gender.male, 1234571).toString() should be ("Chris Shaw")
			generator.generate(Gender.male, 1234572).toString() should be ("Matt Hughes")
			generator.generate(Gender.male, 1234573).toString() should be ("Daniel Edwards")
			generator.generate(Gender.male, 1234574).toString() should be ("Martin Taylor")
			generator.generate(Gender.male, 1234575).toString() should be ("Tom Watson-Thompson")
			generator.generate(Gender.male, 1234576).toString() should be ("Alex Brown")
			generator.generate(Gender.female, 1234567).toString() should be ("Emily Baker")
			generator.generate(Gender.female, 1234568).toString() should be ("Karen Greenton")
			generator.generate(Gender.female, 1234569).toString() should be ("Sophie Robertsford")
			generator.generate(Gender.female, 1234570).toString() should be ("Maria Patel")
			generator.generate(Gender.female, 1234571).toString() should be ("Rebecca Shaw")
			generator.generate(Gender.female, 1234572).toString() should be ("Alison Hughes")
			generator.generate(Gender.female, 1234573).toString() should be ("Lucy Edwards")
			generator.generate(Gender.female, 1234574).toString() should be ("Caroline Taylor")
			generator.generate(Gender.female, 1234575).toString() should be ("Charlotte Watson-Thompson")
			generator.generate(Gender.female, 1234576).toString() should be ("Anna Brown")
		}
	}

}

File: src/test/kotlin/uk/co/halfninja/randomnames/EnglishlikeNameGeneratorTest.kt
package uk.co.halfninja.randomnames

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class EnglishlikeNameGeneratorTest {

    /**
     * Not actually testing anything here.. just printing things out.
     * Fun!
     */
    @Test
    fun `generation of names works`() {
        val generator = NameGenerators.standardGenerator()
        repeat(10) {
            println(generator.generate(Gender.Male))
            println(generator.generate(Gender.Female))
        }
    }

    @Test
    fun `seeded name generation returns expected names`() {
        val generator = NameGenerators.standardGenerator()
        assertEquals("Matthew Baker", generator.generate(Gender.Male, 1234567).toString())
        assertEquals("Sam Greenton", generator.generate(Gender.Male, 1234568).toString())
        assertEquals("Richard Robertsford", generator.generate(Gender.Male, 1234569).toString())
        assertEquals("Jack Patel", generator.generate(Gender.Male, 1234570).toString())
        assertEquals("Chris Shaw", generator.generate(Gender.Male, 1234571).toString())
        assertEquals("Matt Hughes", generator.generate(Gender.Male, 1234572).toString())
        assertEquals("Daniel Edwards", generator.generate(Gender.Male, 1234573).toString())
        assertEquals("Martin Taylor", generator.generate(Gender.Male, 1234574).toString())
        assertEquals("Tom Watson-Thompson", generator.generate(Gender.Male, 1234575).toString())
        assertEquals("Alex Brown", generator.generate(Gender.Male, 1234576).toString())
        assertEquals("Emily Baker", generator.generate(Gender.Female, 1234567).toString())
        assertEquals("Karen Greenton", generator.generate(Gender.Female, 1234568).toString())
        assertEquals("Sophie Robertsford", generator.generate(Gender.Female, 1234569).toString())
        assertEquals("Maria Patel", generator.generate(Gender.Female, 1234570).toString())
        assertEquals("Rebecca Shaw", generator.generate(Gender.Female, 1234571).toString())
        assertEquals("Alison Hughes", generator.generate(Gender.Female, 1234572).toString())
        assertEquals("Lucy Edwards", generator.generate(Gender.Female, 1234573).toString())
        assertEquals("Caroline Taylor", generator.generate(Gender.Female, 1234574).toString())
        assertEquals("Charlotte Watson-Thompson", generator.generate(Gender.Female, 1234575).toString())
        assertEquals("Anna Brown", generator.generate(Gender.Female, 1234576).toString())
    }
}
